core/cas-server-core-tickets-api/src/main/java/org/apereo/cas/ticket/registry/DefaultTicketRegistryCleaner.java

```diff
@@ -45,16 +46,9 @@ public int clean() {
     @Override
     public int cleanTicket(final Ticket ticket) {
         return lockRepository.execute(ticket.getId(), () -> {
-            try {
-                if (ticket instanceof final TicketGrantingTicket tgt) {
-                    LOGGER.debug("Cleaning up expired ticket-granting ticket [{}]", ticket.getId());
-                    val request = SingleLogoutExecutionRequest.builder()
-                        .ticketGrantingTicket(tgt)
-                        .build();
-                    logoutManager.performLogout(request);
-                }
-            } catch (final Throwable e) {
-                LoggingUtils.error(LOGGER, e);
+            if (ticket instanceof final TicketGrantingTicket tgt) {
+                applicationContext.publishEvent(new CasRequestSingleLogoutEvent(this, tgt, null));
```

Comment:
Probably should not pass null here. Pass the actual client info.

Reply:
The cleaner is a non-web process running in background. `ClientInfoHolder.getClientInfo()` returns `null`.

Reply:
I don't think it's worth passing it, isn't it?

Reply:
True, but if the thing expects a client-info, and client-info might (for now) return null, and you're passing null, you might as well pass `ClientInfoHolder.getClientInfo()` anyway, so that when we at some point in the future make this `ClientInfoHolder.getClientInfo()` return a non-null object, no extra work is required.

Reply:
The argument indeed works both ways. I will change.